src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/DiscardSyntaxClassifier.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Classification.Classifiers;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.Classification.Classifiers
+{
+    internal class DiscardSyntaxClassifier : AbstractSyntaxClassifier
+    {
+        public override ImmutableArray<Type> SyntaxNodeTypes { get; } = ImmutableArray.Create(
+            typeof(DiscardDesignationSyntax),
+            typeof(DiscardPatternSyntax),
+            typeof(LambdaExpressionSyntax),
+            typeof(IdentifierNameSyntax));
+
+        public override void AddClassifications(
+           Workspace workspace,
+           SyntaxNode syntax,
+           SemanticModel semanticModel,
+           ArrayBuilder<ClassifiedSpan> result,
+           CancellationToken cancellationToken)
+        {
+            if (syntax.IsKind(SyntaxKind.DiscardDesignation) || syntax.IsKind(SyntaxKind.DiscardPattern))
+            {
+                result.Add(new ClassifiedSpan(syntax.Span, ClassificationTypeNames.Keyword));
+                return;
+            }
+
+            var symbolInfo = semanticModel.GetSymbolInfo(syntax, cancellationToken);
+
+            switch (syntax)
+            {
+                case LambdaExpressionSyntax _:
+                    ClassifyLambdaParameter(symbolInfo, result);
+                    break;
+
+                default:
```

Comment:
add `Debug.Assert(syntax is IdentifierNameSyntax)`

Reply:
done in the default now since i am now casing/matching IdentifierNameSyntax. Should the assert have a message or nah?

Reply:
that would also be fine with me.